tests/cbmc/proofs/s2n_strcpy/s2n_strcpy_harness.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_str.h"
+
+#include <assert.h>
+#include <string.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <sys/param.h>
+
+void s2n_strcpy_harness() {
+    char *str = ensure_c_str_is_allocated(MAX_STRING_LEN);
+    const uint32_t slen = strlen(str);
+    const uint32_t buflen;
+    const uint32_t index;
+    __CPROVER_assume(buflen < MAX_STRING_LEN);
+    __CPROVER_assume(index < strlen(str));
```

Comment:
use `slen` to avoid recalculating.
Also, you don't want to block if `slen == 0`
```suggestion
    __CPROVER_assume(slen == 0 || index < slen);
```